### ISSUE TITLE
WebAssemply and GitHub Pages

### ISSUE THREAD

**Issue by homersimpsons** [mistricky/CodeSnap#44]
To ease the usage and avoid requiring installation it could be great if this tool worked directly in the web.

It should be possible to push it to [GitHub Pages](https://pages.github.com/) if it can be built into WebAssembly.

Thank you for your work on this.

**Comment by mistricky** [comment]
@homersimpsons WebApp indeed does not require installation, but it needs users to paste their code into the WebApp, CLI may not be the most convenient way to use it at this moment, but I'll provide more usage options in the future, like editor plugin or something, so that users only need to select the code they want to generate, and press some hotkey, the code snapshot will paste it into clipboard or save in somewhere.

**Comment by homersimpsons** [comment]
> but it needs users to paste their code into the WebApp

Yes, but I believe it is more convenient to use to quickly create a share the image. For example, I think I will use CodeSnap for some slides or post thumbnail, but I do not want to install one more thing and to remember how to use it and to keep it up-to-date.

Having a basic GitHub pages with a textarea input and a generate button will be way more convenient for one-off usage or for sparse usages.

> but I'll provide more usage options in the future, like editor plugin or something, so that users only need to select the code they want to generate, and press some hotkey, the code snapshot will paste it into clipboard or save in somewhere.

This feel far more complicated than having a GitHub page to allow a copy-paste.

I can try to take a look to having this.

**Comment by mistricky** [comment]
@homersimpsons now only https://github.com/mistricky/codesnap.nvim is available, if you are using Neovim you can try it. And I will provide VSCode and Zed plugin in the future.

CodeSnap also provide library to create a code snapshot tool, so if you want WebApp, you can build your own based on CodeSnap 😉

**Comment by mistricky** [comment]
Now I have release a wasm package called CodeSnap.wasm
